lib/gui/models/store.js

```diff
@@ -94,15 +95,15 @@ const storeReducer = (state, action) => {
 
         const drive = _.first(action.data);
 
-        if (_.every([
-
-          // TODO: Reuse from SelectionStateModel.isDriveValid()
-          state.getIn([ 'selection', 'image', 'size' ], 0) <= drive.size,
+        // Even if there's no image selectected, we need to call `isDriveValid`
+        // to check if the drive is locked, and `{}` works fine with it
+        const image = state.getIn([ 'selection', 'image' ], {
```

Comment:
I wonder if just for clarity, we should let this be `undefined` if it doesn't exist, and move the "empty object" workaround to the places that actually need it:

```js
if (_.every([
    constraints.isDriveValid(drive, image || {}),
    constraints.isDriveSizeRecommended(drive, image || {})
 ]))
```

This would prevent us from accidentally relying on an empty object representing the lack of an image in the store.